### PR TITLE
More unifications and adjustments

### PR DESCRIPTION
**Pull request by guerler**
Remove redundancies from preference main view, decouple sub views from main view. Fix naming inconsistencies in front and backend. Unify view handling.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/user_preferences.py [hunk 1]
```diff
@@ -783,14 +798,7 @@ def get_filter_mapping(db_filters, config_filters, factory):
...
     @expose_api
-    def toolbox_filters(self, trans, cntrller='user_preferences', **kwd):
-        """
-        API call for fetching toolbox filters data
-        """
-        return self.tool_filters(trans, **kwd)
-
-    @expose_api
-    def edit_toolbox_filters(self, trans, cntrller='user_preferences', **kwd):
+    def __edit_toolbox_filters(self, trans, cntrller='user_preferences', **kwd):
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/user_preferences.py:801]: I don't think something with an `@expose_api` tag should be private like this. If it is unused now I'd just remove it.

File: lib/galaxy/webapps/galaxy/buildapp.py [hunk 2]
```diff
@@ -488,28 +494,28 @@ def populate_api_routes( webapp, app ):
...
+                           action='change_permissions',
                            conditions=dict( method=["GET"] ) )
 
-    webapp.mapper.connect( 'api_keys',
-                           '/api/user_preferences/api_keys',
+    webapp.mapper.connect( 'change_api_key',
+                           '/api/user_preferences/change_api_key',
                            controller='user_preferences',
-                           action='api_keys',
+                           action='change_api_key',
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/buildapp.py:506]: I also don't get `change_api_key` shouldn't it be a `PUT` on `api_keys`? Some of these endpoints seemed more RESTful before this change.

File: lib/galaxy/webapps/galaxy/api/user_preferences.py (hunk 1 above)
Comment (guerler) [reply lib/galaxy/webapps/galaxy/api/user_preferences.py:801]: This is a work in progress and in transition. Edit toolbox filters e.g. should either be part of change_toolbox_filters, or split into set/get_toolbox_filters. If change_toolbox_filters does not sound good we can change the name in a consistent fashion.

File: lib/galaxy/webapps/galaxy/buildapp.py (hunk 2 above)
Comment (guerler) [reply lib/galaxy/webapps/galaxy/buildapp.py:506]: Same here, we could name this endpont either, edit_xxx, change_xxx, manage_xxx or just xxx, as long as its consistent, but this is in transition. As above the goal is to have consistent endpoint naming. I would prefer to split it by methods get and put but we are not there yet.

**Comment by bgruening** [comment]
Thanks @guerler and @jmchilton for the review. I hope we can finish this off pretty soon and call for a new review!

**Comment by guerler** [comment]
@jmchilton thx for looking into it. I am on board renaming them all without 'change' for the next iteration if thats better for now I just wanted them to be name consistently for starters.

**Pull request merged**